sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -465,5 +468,193 @@ describe(
+              },
+              calculatedData: {
+                familySize: 2,
+                studentMSOLAllowance: 7777,
+                totalNonEducationalCost: 22,
+                studentMaritalStatusCode: "SI",
+                pdppdStatus: true,
+              },
+            },
+          },
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+
+      // Queued job.
+      const { job } = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processECert(job);
+
+      // Assert 0 uploaded records.
+      expect(result).toStrictEqual([
+        "Process finalized with success.",
+        "Generated file: none",
+        "Uploaded records: 0",
+      ]);
+    });
+    it("Should generate disbursement if the Student assessment contains PD/PPD application flag=yes and Student profile PD approved", async () => {
+      // Arrange
+
+      // Student with valid SIN.
+      const student = await saveFakeStudent(
+        db.dataSource,
+        {},
+        {
+          initialValue: {
+            disabilityStatus: DisabilityStatus.PD,
+          },
+        },
+      );
+      // Valid MSFAA Number.
+      const msfaaNumber = await db.msfaaNumber.save(
+        createFakeMSFAANumber({ student }, { msfaaState: MSFAAStates.Signed }),
+      );
+      // Student application eligible for e-Cert.
+      await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student, msfaaNumber },
+        {
+          offeringIntensity: OfferingIntensity.fullTime,
+          applicationStatus: ApplicationStatus.Completed,
+          currentAssessmentInitialValues: {
+            assessmentData: { weeks: 5 } as Assessment,
+            assessmentDate: new Date(),
+            workflowData: {
+              studentData: {
+                dependantStatus: "dependant",
+                relationshipStatus: RelationshipStatus.Single,
+                livingWithParents: FormYesNoOptions.Yes,
+                numberOfParents: 2,
+              },
+              calculatedData: {
+                familySize: 2,
+                studentMSOLAllowance: 7777,
+                totalNonEducationalCost: 22,
+                studentMaritalStatusCode: "SI",
+                pdppdStatus: true,
+              },
+            },
+          },
+          firstDisbursementInitialValues: {
+            coeStatus: COEStatus.completed,
+          },
+        },
+      );
+
+      // Queued job.
+      const { job } = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processECert(job);
+
+      // Assert uploaded file.
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const fileDate = dayjs().format("YYYYMMDD");
+      const uploadedFileName = `MSFT-Request\\DPBC.EDU.FTECERTS.${fileDate}.001`;
+      expect(uploadedFile.remoteFilePath).toBe(uploadedFileName);
+      expect(result).toStrictEqual([
```

Comment:
We can check the db for the disbursement id to make sure that disbursement schedule status is Pending.